[ENH] Moved test skip config for DistanceFeatures to estimator tags

IsaiasRT opened the pull request:
**Reference Issues/PRs**
- https://github.com/sktime/sktime/issues/8515

What does this implement/fix?
It migrates test skip logic for the DistanceFeatures estimator from the centralized tests/_config.py file into the estimator itself using the _tags compose_distance.py

fkiraly commented on sktime/tests/_config.py:
```diff
@@ -34,6 +34,8 @@
 # DO NOT ADD ESTIMATORS HERE ANYMORE
 # ADD TEST SKIPS TO TAG tag tests:skip_all INSTEAD
 EXCLUDE_ESTIMATORS = [
+    # SFA is non-compliant with any transformer interfaces, #2064
```
the comments should be moved to `SFA` too

fkiraly commented on sktime/transformations/panel/dictionary_based/_sfa.py:
```diff
@@ -115,6 +115,8 @@ class SFA(BaseTransformer):
         "X_inner_mtype": "numpy3D",  # which mtypes do _fit/_predict support for X?
         "y_inner_mtype": "pd_Series_Table",  # which mtypes does y require?
         "requires_y": True,  # does y need to be passed in fit?
+        "python_dependencies": ["numba", "joblib"],
```
this is a duplicate line, remove

fkiraly reviewed (changes requested):
Thanks! But SFA has already been dealt with, there was nothing to move.

fkiraly commented:
Ah, I think I understand what is going on. You probably did not update your fork!

So, SFA was still present in your older version of `sktime`!

If you look at the difference, you see you are actually *adding back some of the comments* in overall effect, instead of moving things to the estimator!
https://github.com/sktime/sktime/pull/8520/files

Learning moment: always keep your fork updated!
(in the GitHub GUI, it is the "sync fork" button on top right of the web interface of your fork)

fkiraly commented:
Great! You can update the original post and title of this PR by going on the three dots at the top right, and "edit" button next to the title.

dtemir commented on sktime/tests/_config.py:
```diff
@@ -31,9 +31,9 @@
 
-# DO NOT ADD ESTIMATORS HERE ANYMORE
-# ADD TEST SKIPS TO TAG tag tests:skip_all INSTEAD
 EXCLUDE_ESTIMATORS = [
+    # SFA is non-compliant with any transformer interfaces, #2064
+    "SFA",
     # PlateauFinder seems to be broken, see #2259
```
these should go back.

you shouldn't delete the warning "DO NOT ADD ESTIMATORS HERE ANYMORE"

and you shouldn't put the "SFA" back since that one was resolved

dtemir commented on sktime/tests/_config.py:
```diff
@@ -87,8 +87,8 @@
     "RecursiveReductionForecaster",
-    # DistanceFeatures does ont work for hierarchical data, see #8077
-    "DistanceFeatures",
+    # DistanceFeatures does not work for hierarchical data, see #8077
+    #removed DistanceFeatures
```
i think you should just delete the mention of `DistanceFeatures` from these lines

dtemir commented on sktime/transformations/panel/dictionary_based/_sfa.py:
```diff
@@ -115,6 +115,8 @@ class SFA(BaseTransformer):
         "X_inner_mtype": "numpy3D",  # which mtypes do _fit/_predict support for X?
         "y_inner_mtype": "pd_Series_Table",  # which mtypes does y require?
         "requires_y": True,  # does y need to be passed in fit?
+        "python_dependencies": ["numba", "joblib"],
+        "tests:skip_all": True #added entry test:skip tag
```
you shouldn't have to modify the `_sfa.py` file anymore, right? since you're no longer working on SFA, but focusing on `compose_distance.py`

dtemir replied:
```suggestion
        # DistanceFeatures does ont work for hierarchical data, see #8077
        "tests:skip_all": True
```

dtemir commented on sktime/transformations/panel/compose_distance.py:
```diff
@@ -70,6 +70,7 @@ class DistanceFeatures(BaseTransformer):
         # CI and test flags
         # -----------------
         "tests:core": True,  # should tests be triggered by framework changes?
+        "tests:skip_all": True, #test:skip added
```
```suggestion
        # DistanceFeatures does ont work for hierarchical data, see #8077
        "tests:skip_all": True, #test:skip added
```

IsaiasRT commented:
**Restore docstrings in config.py, revert SFA to original, and apply dtemir suggestion to compose_distance.py**

- Reinstated missing docstrings in config.py
- Reverted changes made to sfa.py to preserve original implementation
- Incorporated suggested updates from dmetir to compose_distance.py

dtemir commented on sktime/tests/_config.py:
```diff
@@ -30,10 +30,9 @@
 # DO NOT ADD ESTIMATORS HERE ANYMORE
-# ADD TEST SKIPS TO TAG tag tests:skip_all INSTEAD
 EXCLUDE_ESTIMATORS = [
+    # SFA is non-compliant with any transformer interfaces, #2064
```
```suggestion
    
```

fkiraly commented on sktime/tests/_config.py:
```diff
@@ -30,10 +30,9 @@
 # DO NOT ADD ESTIMATORS HERE ANYMORE
-# ADD TEST SKIPS TO TAG tag tests:skip_all INSTEAD
 EXCLUDE_ESTIMATORS = [
+    
```
for example, here

fkiraly commented on sktime/tests/_config.py:
```diff
@@ -103,8 +100,6 @@
 
 
-# DO NOT ADD ESTIMATORS HERE ANYMORE
-# ADD TEST SKIPS TO TAG tag tests:skip_by_name INSTEAD
```
for example, here

fkiraly commented on sktime/tests/_config.py:
```diff
@@ -118,6 +113,8 @@
     # known side effects on multivariate arguments, #2072
     "WindowSummarizer": ["test_methods_have_no_side_effects"],
+    # test fails in the Panel case for Differencer, see #2522
+    "Differencer": ["test_transform_inverse_transform_equivalent"],
```
for example, here

fkiraly commented:
Thanks - I think at this point in time it is perhaps best to clean up the following way:

* change your branch to `main`
* copy the `_config.py` file somewhere
* switch back to your branch
* copy the saved `_config.py` over the one in the branch
* re-apply your changes

fkiraly reviewed (changes requested):
Great! Looks like this worked. You should also revert the `_sfa` file.

dtemir commented on sktime/transformations/panel/dictionary_based/_sfa.py:
```diff
@@ -118,7 +118,6 @@ class SFA(BaseTransformer):
         # testing configuration
         # ---------------------
         # SFA is non-compliant with any transformer interfaces, #2064
-        "tests:skip_all": True,
```
this change still remains to be put back

fkiraly replied:
no problem! As already said, we intentionally did pick this task that has a relatively simple scope in python, since it helps getting familiar with GitHub! I hope I did not promise too much 😃 

Let us know if you need any help with "how to do x or y". (discord/dev-chat)

fkiraly reviewed (approved):
ok, looks good now!
